### PR TITLE
Simplify the element methods

### PR DESCRIPTION
**Pull request by craigbr**
Move repeated logic into client.go

Restructure post, put, patch methods to reuse code

### CODE REVIEW

File: monascaclient/client.go [hunk 1]
```diff
@@ -160,6 +161,63 @@ func (c *Client) callMonascaNoContent(monascaURL string, method string, requestB
...
+	return nil
+}
+
+func (c *Client) callMonascaWithBody(basePath string, id string, method string, toSend interface{}, returned interface{}) error {
+	path := basePath
+	if id != "" {
+		path = path + "/" + id
+	}
+	monascaURL, URLerr := c.createMonascaAPIURL(path, nil)
+	fmt.Printf("MonascaURL = %s", monascaURL)
```
Comment (mhoppal) [inline monascaclient/client.go]: take out

Comment (mhoppal) [reply monascaclient/client.go]: the print

**Comment by craigbr** [comment]
Done

File: monascaclient/notifications.go [hunk 2]
```diff
@@ -44,136 +47,48 @@ func DeleteNotificationMethod(notificationID string) error {
...
-	}
-
-	body, monascaErr := c.callMonascaReturnBody(monascaURL, "GET", nil)
-	if monascaErr != nil {
-		return nil, monascaErr
-	}
-
-	notificationMethodElement := models.NotificationElement{}
-	err := json.Unmarshal(body, &notificationMethodElement)
+	path := notificationsBasePath + "/" + notificationMethodID
```
Comment (mhoppal) [inline monascaclient/notifications.go]: couldnt you do this in the method much like you do with callMonascaWithBody

**Review by mhoppal — APPROVED** [review]
Awesome work! 👍

**Pull request merged**